src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -57,27 +58,27 @@ internal abstract partial class HttpProtocol : IHttpResponseControl
         private bool _hasAdvanced;
         private bool _isLeasedMemoryInvalid = true;
         private bool _autoChunk;
-        protected Exception _applicationException;
-        private BadHttpRequestException _requestRejectedException;
+        protected Exception? _applicationException;
+        private BadHttpRequestException? _requestRejectedException;
 
         protected HttpVersion _httpVersion;
         // This should only be used by the application, not the server. This is settable on HttpRequest but we don't want that to affect
         // how Kestrel processes requests/responses.
-        private string _httpProtocol;
+        private string? _httpProtocol;
 
-        private string _requestId;
+        private string? _requestId;
         private int _requestHeadersParsed;
 
         private long _responseBytesWritten;
 
-        private HttpConnectionContext _context;
-        private RouteValueDictionary _routeValues;
-        private Endpoint _endpoint;
+        private HttpConnectionContext _context = default!; // Always initialized with a value
```

Comment:
I don't think we need the `// Always initialized with a value` comments for `= default!`. I think it's clear enough. I noticed this is inconsistant anyway. ResponseHeaders doesn't have this comment for instance.